src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -236,6 +247,26 @@ export class DeploymentsService implements OnDestroy {
         deepEqual(new Set<string>(p.map(v => v.name)), new Set<string>(q.map(v => v.name))));
   }
 
+  getAppAndEnvironments(spaceId: string): Observable<ApplicationAttributesOverview[]> {
+    return this.getApplicationsResponse(spaceId)
+      .map((apps: Application[]) => apps || [])
+      .map((apps: Application[]) => apps.map((app: Application) => {
+        let appName = app.attributes.name;
+        let deploymentNamesAndVersions = app.attributes.deployments.map(
+          (dep: Deployment) => { return {
+            name: dep.attributes.name,
+            version: dep.attributes.version,
+            url: dep.links.application
+          }; }
```

Comment:
you can do `(dep: Deployment) => ({ name: dep.attributes.name, version: dep.attributes.version, url: dep.links.application })` and save some of the visual clutter here